test/framework/src/main/java/org/elasticsearch/test/AbstractQueryTestCase.java

```diff
@@ -187,14 +187,10 @@ public static void beforeClass() {
 
         index = new Index(randomAlphaOfLengthBetween(1, 10), "_na_");
 
-        // Set a single type in the index
-        switch (random().nextInt(3)) {
-        case 0:
+        if (rarely()) {
             currentTypes = new String[0]; // no types
-            break;
-        default:
+        } else {
             currentTypes = new String[] { "_doc" };
```

Comment:
If we only have one allowed type on master now, we could also look into making currentTypes a String, not an array. also changing it to "currentType" might be better then.